sdk/clientcore/annotation-processor/src/main/java/io/clientcore/annotation/processor/templating/JavaParserTemplateProcessor.java

```diff
@@ -341,69 +343,160 @@ void initializeHttpRequest(BlockStmt body, HttpRequestContext method) {
+                    // Handle non-String values (e.g., int, boolean, etc.)
+                    body.addStatement("queryParamMap.put(\"" + key + "\", " + valueExpression + ");");
                 }
-            });
-            body.addStatement("newUrl = CoreUtils.appendQueryParams(url, queryParamMap);");
-            body.addStatement("if (newUrl != null) { url = newUrl; }");
-        }
+            } else {
+                // Handle null or empty values
+                body.addStatement("queryParamMap.put(\"" + key + "\", null);");
+            }
+        });
+
+        // Append query parameters to the URL
+        body.addStatement("String newUrl = CoreUtils.appendQueryParams(url, queryParamMap);");
+        body.addStatement("if (newUrl != null) { url = newUrl; }");
     }
 
     private void addHeadersToRequest(BlockStmt body, HttpRequestContext method) {
         if (method.getHeaders().isEmpty()) {
             return;
         }
 
-        body.tryAddImportToParentCompilationUnit(HttpHeaderName.class);
         StringBuilder httpRequestBuilder = new StringBuilder("httpRequest.getHeaders()"); // Start the header chaining
+
         for (Map.Entry<String, String> header : method.getHeaders().entrySet()) {
+            String headerKey = header.getKey();
+            String headerValue = header.getValue();
+
+            // Determine if the header value is a String type
             boolean isStringType = method.getParameters()
                 .stream()
-                .anyMatch(parameter -> parameter.getName().equals(header.getValue())
+                .anyMatch(parameter -> parameter.getName().equals(headerValue)
                     && "String".equals(parameter.getShortTypeName()));
-            String value = isStringType ? header.getValue() : "String.valueOf(" + header.getValue() + ")";
-            String constantName
-                = LOWERCASE_HEADER_TO_HTTPHEADENAME_CONSTANT.get(header.getKey().toLowerCase(Locale.ROOT));
+
+            // Split the header value by commas
+            String[] values = headerValue.split(",");
+            String valueExpression;
+
+            if (values.length > 1) {
+                body.tryAddImportToParentCompilationUnit(HttpHeader.class);
+                body.tryAddImportToParentCompilationUnit(Arrays.class);   // Handle multiple values by constructing a List<String>
+                valueExpression = "Arrays.asList(" + Arrays.stream(values).map(value -> {
+                    value = value.trim(); // Trim whitespace
+                    if (value.startsWith("\"")) {
+                        value = value.substring(1); // Remove the starting quote
+                    }
+                    if (value.endsWith("\"")) {
+                        value = value.substring(0, value.length() - 1); // Remove the ending quote
+                    }
+                    return "\"" + value + "\""; // Add quotes around the trimmed value
```

Comment:
If the value starts and ends with `"`, should we just return it as is? We're stripping possible leading and trailing `"` and then adding it right back. Thoughts on using this logic:

- If starts with and ends with `"`, return as-is.
- If starts with `"`, return `value + "\""`.
- If ends with `"`, return `"\"" + value`.
- Return `"\"" + value + "\""`

Reply:
>? We're stripping possible leading and trailing " and then adding it right back.

It is to handle cases where we get the static values with enclosing quotes, for example, "constantValue1;constantValue2" The above logic then seperates each to its quoted value which is,  [ "constantValue1", "constantValue2"]